src/main/java/Dog.java

```diff
@@ -38,8 +39,14 @@ public void doeat(){
     }
 
     public void dowalk(){
-        System.out.println(name+" is walikng");
-        this.walk=true;
+        if(this.eat==true){
+            System.out.println(name+" can't eat because it is eating");
+        }
+        else{
+            System.out.println(name+" is walikng");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)